Content.Server/GameObjects/Components/Weapon/Melee/StunbatonComponent.cs

```diff
@@ -25,195 +25,40 @@
-            }
-
-            var sprite = Owner.GetComponent<SpriteComponent>();
-            var item = Owner.GetComponent<ItemComponent>();
-
-            SoundSystem.Play(Filter.Pvs(Owner), AudioHelpers.GetRandomFileFromSoundCollection("sparks"), Owner.Transform.Coordinates, AudioHelpers.WithVariation(0.25f));
-
-            item.EquippedPrefix = "off";
-            sprite.LayerSetState(0, "stunbaton_off");
-            _activated = false;
-        }
-
-        private void TurnOn(IEntity user)
-        {
-            if (_activated)
-            {
-                return;
-            }
-
-            var sprite = Owner.GetComponent<SpriteComponent>();
-            var item = Owner.GetComponent<ItemComponent>();
-
-            var playerFilter = Filter.Pvs(Owner);
-            if (Cell == null)
-            {
-                SoundSystem.Play(playerFilter, "/Audio/Machines/button.ogg", Owner.Transform.Coordinates, AudioHelpers.WithVariation(0.25f));
-
-                Owner.PopupMessage(user, Loc.GetString("Cell missing..."));
-                return;
-            }
-
-            if (Cell.CurrentCharge < EnergyPerUse)
-            {
-                SoundSystem.Play(playerFilter, "/Audio/Machines/button.ogg", Owner.Transform.Coordinates, AudioHelpers.WithVariation(0.25f));
-                Owner.PopupMessage(user, Loc.GetString("Dead cell..."));
-                return;
-            }
-
-            SoundSystem.Play(playerFilter, AudioHelpers.GetRandomFileFromSoundCollection("sparks"), Owner.Transform.Coordinates, AudioHelpers.WithVariation(0.25f));
-
-            item.EquippedPrefix = "on";
-            sprite.LayerSetState(0, "stunbaton_on");
-            _activated = true;
-        }
-
-        bool IUse.UseEntity(UseEntityEventArgs eventArgs)
-        {
-            ToggleStatus(eventArgs.User);
-
-            return true;
-        }
-
-        async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
-        {
-            if (!ActionBlockerSystem.CanInteract(eventArgs.User)) return false;
-            if (!_cellSlot.InsertCell(eventArgs.Using)) return false;
-            Dirty();
-            return true;
-        }
-
-        public void Examine(FormattedMessage message, bool inDetailsRange)
-        {
-            if (Activated)
-            {
-                message.AddMarkup(Loc.GetString("The light is currently [color=darkgreen]on[/color]."));
-            }
-        }
-
-        void IThrowCollide.DoHit(ThrowCollideEventArgs eventArgs)
-        {
-            if (!Activated || Cell == null || !Cell.TryUseCharge(EnergyPerUse) || !eventArgs.Target.TryGetComponent(out StunnableComponent? stunnable))
-                return;
-
-            SoundSystem.Play(Filter.Pvs(Owner), "/Audio/Weapons/egloves.ogg", Owner.Transform.Coordinates, AudioHelpers.WithVariation(0.25f));
-
-            stunnable.Paralyze(_paralyzeTime);
+            CellSlot = Owner.EnsureComponent<PowerCellSlotComponent>();
```

Comment:
Never ensure components, always `TryGetComponent` and default to not doing anything if the component is not there.